paper-api/src/main/java/org/bukkit/entity/HumanEntity.java

```diff
@@ -701,11 +703,55 @@ default void openSign(@NotNull org.bukkit.block.Sign sign) {
      * This will force the entity to drop the item they are holding with
      * an option to drop the entire {@link ItemStack} or just 1 of the items.
      *
+     * @deprecated You should instead use {@link #dropItem(EquipmentSlot, int)} with a {@link EquipmentSlot#HAND} parameter.
      * @param dropAll True to drop entire stack, false to drop 1 of the stack
      * @return True if item was dropped successfully
      */
+    @Deprecated(since = "1.21.4")
     public boolean dropItem(boolean dropAll);
 
+    /**
+     * Makes the entity drop an item from their inventory based on the slot.
+     *
+     * @param slot   The slot to drop
+     * @param amount The number of items to drop from this slot. Values below 1 don't drop an Item
+     * @return The dropped item entity, or null if the action was unsuccessful
+     * @throws IllegalArgumentException If the slot is negative or bigger than the player's inventory
+     */
+    public @Nullable Item dropItem(int slot, int amount);
+
+    /**
+     * Makes the player drop an item from their inventory based on the equipment slot.
+     *
+     * @param slot   The equipment slot to drop
+     * @param amount The amount of items to drop from this equipment slot. Values below 1 don't drop an Item
+     * @return The dropped item entity, or null if the action was unsuccessful
+     */
+    public @Nullable Item dropItem(@NotNull EquipmentSlot slot, int amount);
+
+    /**
+     * Makes the entity drop an item from their inventory based on the slot.
+     * Instead of the item being dropped where the player is currently looking, this method makes it drop in
+     * a random direction, similar to how items are dropped after a player's death.
+     *
+     * @param slot   The slot to drop
+     * @param amount The number of items to drop from this slot. Values below 1 don't drop an Item
+     * @return The dropped item entity, or null if the action was unsuccessful
+     * @throws IndexOutOfBoundsException If the slot is negative or bigger than the player's inventory
```

Comment:
forgot about this IllegalArgumentException

Reply:
Woops